pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -545,6 +547,18 @@ func (rc *reconciler) updateDevicePath(volumesNeedUpdate map[v1.UniqueVolumeName
 	} else {
 		for _, attachedVolume := range node.Status.VolumesAttached {
 			if volume, exists := volumesNeedUpdate[attachedVolume.Name]; exists {
+				// If volume devicePath is already known from previous mount checking, no need to update it again.
+				if volume.devicePath != "" {
+					target, err := filepath.EvalSymlinks(attachedVolume.DevicePath)
+					if err != nil {
+						target = attachedVolume.DevicePath
```

Comment:
Where is `attachedVolume.DevicePath` set?

Reply:
The attachedVolumes are obtained from node status VolumesAttached. The devicePath are set by the volume controller in master.

Reply:
Double check what happens if `attachedVolume.DevicePath` is empty

Reply:
added the check